exercises/say/src/example.c

```diff
@@ -0,0 +1,141 @@
+#include "say.h"
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <math.h>
+
+typedef struct {
+   const char *name;
+   size_t size;
+} map_t;
+
+static const map_t below_10[] = {
+   {"", 0}, {"one", 3}, {"two", 3}, {"three", 5},
+   {"four", 4}, {"five", 4}, {"six", 3}, {"seven", 5},
+   {"eight", 5}, {"nine", 4}
+};
+
+static const map_t below_20[] = {
+   {"ten", 3}, {"eleven", 6}, {"twelve", 6}, {"thirteen", 8},
+   {"fourteen", 8}, {"fifteen", 7}, {"sixteen", 7}, {"seventeen", 9},
+   {"eighteen", 8}, {"nineteen", 8}
+};
+
+static const map_t below_100[] = {
+   {"", 0}, {"", 0}, {"twenty", 6}, {"thirty", 6},
+   {"forty", 5}, {"fifty", 5}, {"sixty", 5}, {"seventy", 7},
+   {"eighty", 6}, {"ninety", 6}
+};
+
+static const map_t thousands[] = {
+   {"", 0}, {"thousand", 8}, {"million", 7}, {"billion", 7}
+};
+
+/* below_20's are like two times below_10's so this will work */
+static const int widest = 8;
+
+int say(int64_t input, char **ans)
+{
+   if (input < 0 || input > 999999999999 || ans == NULL)
+      return -1;
+
+   /* 'zero' is a corner case */
+   if (input == 0) {
+      *ans = malloc(5);
```

Comment:
How about `*ans = "zero";` instead?

Reply:
I think this will require you to change to `const char **ans`

Reply:
How can I then build the result if it is ```const```?

Reply:
It is a pointer to const data and the compiler will replace a string literal with a `const void *` that points to the constant string data.

Reply:
I think I understand now what you're saying: the client needs to be able to `free` this memory so we can't declare it as `const`. I agree with you.

Reply:
Good point.
My question was how to use ```strcpy``` when the destination string is a ```const```? Build the result in a modifiable char and then cast the return to ```const char```?

Reply:
Well, in this case I was suggesting that you _not_ use `strcpy` but instead just set `*ans` to a string literal. This would work except that it would not be possible to free the string.

Reply:
Got it - thanks for the clarification!